### PR TITLE
feat(postage): network commitment

### PR DESCRIPTION
**Pull request by acud**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2767)

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

**Review by mrekucci — CHANGES_REQUESTED** [review]
> Reviewed 7 of 8 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2767)* status: all files reviewed, 1 unresolved discussion (waiting on @acud)

___
*[pkg/debugapi/postage.go, line 134 at r2](https://reviewable.io/reviews/ethersphere/bee/2767#-MtgbBLk0Ur1b3_KgR1R:-MtgbBLl0pfndX5IV5yS:b-sscaoo) ([raw file](https://github.com/ethersphere/bee/blob/92d3b5581b42a64837314cafe7d64ad3aba445e3/pkg/debugapi/postage.go#L134)):*
> ```Go
> type postageBatchResponse struct {
> 	BatchID     hexByte  `json:"batchID"`
> 	Value       *big.Int `json:"value"`
> ```

We're using `*bigint.BigInt` in JSON responses.

File: pkg/debugapi/postage.go [hunk 1]
```diff
@@ -284,10 +332,23 @@ type chainStateResponse struct {
 func (s *Service) reserveStateHandler(w http.ResponseWriter, _ *http.Request) {
 	state := s.batchStore.GetReserveState()
 
+	commitment := int64(0)
+	if err := s.batchStore.Iterate(func(b *postage.Batch) (bool, error) {
+		commitment += int64(math.Pow(2.0, float64(b.Depth)))
```
Comment (aloknerurkar) [inline pkg/debugapi/postage.go:337]: Does this iteration only happen on valid batches?... do we need to check if the `(normalised balance > cumulative payout)`?

Comment (acud) [reply pkg/debugapi/postage.go:337]: yes and no we don't... the batchstore should take care that all expired batches get _removed_ from the batchstore, therefore resulting in no iteration over those batches.

**Review by mrekucci — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2767)* status: all files reviewed, 2 unresolved discussions (waiting on @acud and @aloknerurkar)

**Pull request merged**